### PR TITLE
WIP Preparing release version 5.0.0

### PR DESCRIPTION
**Pull request by nicoddemus**

### CODE REVIEW

File: doc/en/example/pythoncollection.rst [hunk 1]
```diff
@@ -211,12 +211,37 @@ You can always peek at the collection tree without running tests like this:
...
+    INTERNALERROR>     items = self._perform_collect(args, genitems)
+    INTERNALERROR>   File "$PYTHON_PREFIX/lib/python3.6/site-packages/_pytest/main.py", line 479, in _perform_collect
+    INTERNALERROR>     parts = self._parsearg(arg)
+    INTERNALERROR>   File "$PYTHON_PREFIX/lib/python3.6/site-packages/_pytest/main.py", line 647, in _parsearg
+    INTERNALERROR>     parts[0] = self._tryconvertpyarg(parts[0])
+    INTERNALERROR>   File "$PYTHON_PREFIX/lib/python3.6/site-packages/_pytest/main.py", line 633, in _tryconvertpyarg
+    INTERNALERROR>     spec = importlib.util.find_spec(x)
+    INTERNALERROR>   File "$PYTHON_PREFIX/lib/python3.6/importlib/util.py", line 89, in find_spec
+    INTERNALERROR>     return _find_spec(fullname, parent.__path__)
+    INTERNALERROR> AttributeError: module 'pythoncollection' has no attribute '__path__'
```
Comment (nicoddemus) [inline doc/en/example/pythoncollection.rst]: Hmmm... weird. Any ideas @asottile? 🤔

Comment (asottile) [reply doc/en/example/pythoncollection.rst]: Weird, I'll look at this when I'm at a computer. Seems bad, lol (almost certainly a quirk with the pyargs code, which probably has fewer tests than it needs)

**Comment by asottile** [comment]
Here is the minimal reproduction, I'll work on getting a fix for this:

```bash
touch conftest.py t.py && pytest --pyargs t.py
```

It's interpreting `t.py` as a module and then trying to `import t.py` (which rightfully errors)

File: doc/en/announce/release-5.0.0.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+* Pulkit Goyal
+* Ralph Giles
+* Ronny Pfannschmidt
+* Thomas Grainger
+* Thomas Hisch
+* Tim Gates
+* Victor Maryama
+* Yuri Apollov
+* Zac Hatfield-Dodds
+* Zac-HD
```
Comment (Zac-HD) [inline doc/en/announce/release-5.0.0.rst]: Looks like I have some bad git metadata somewhere - we can delete this line.

**Comment by asottile** [comment]
#5503 fixes the INTERNALERROR

File: CHANGELOG.rst [hunk 3]
```diff
@@ -18,6 +18,188 @@ with advance notice in the **Deprecations** section of releases.
...
+- `#1149 <https://github.com/pytest-dev/pytest/issues/1149>`_: Pytest no longer accepts prefixes of command-line arguments, for example
+  typing ``pytest --doctest-mod`` inplace of ``--doctest-modules``.
+  This was previously allowed where the ``ArgumentParser`` thought it was unambiguous,
+  but this could be incorrect due to delayed parsing of options for plugins.
+  See for example issues `#1149 <https://github.com/pytest-dev/pytest/issues/1149>`__,
+  `#3413 <https://github.com/pytest-dev/pytest/issues/3413>`__, and
+  `#4009 <https://github.com/pytest-dev/pytest/issues/4009>`__.
+
+
+- `#5125 <https://github.com/pytest-dev/pytest/issues/5125>`_: ``Session.exitcode`` values are now coded in ``pytest.ExitCode``, an ``IntEnum``. This makes the exit code available for consumer code and are more explicit other than just documentation. User defined exit codes are still valid, but should be used with caution.
```
Comment (nicoddemus) [inline CHANGELOG.rst]: Note: this should be a feature.

File: CHANGELOG.rst [hunk 4]
```diff
@@ -18,6 +18,188 @@ with advance notice in the **Deprecations** section of releases.
 
 .. towncrier release notes start
 
+pytest 5.0.0 (2019-06-26)
+=========================
+
```
Comment (nicoddemus) [inline CHANGELOG.rst]: Note: add an announcement about this being a Python 3.5+ only release.

File: CHANGELOG.rst (hunk 3 above)
Comment (asottile) [reply CHANGELOG.rst]: hmmm, the old constants were removed as well so perhaps this is in the correct place?

Comment (asottile) [reply CHANGELOG.rst]: ah but they were not publically exposed, so all good :+1:

Comment (nicoddemus) [reply CHANGELOG.rst]: Exactly :)

**Pull request closed without merging**